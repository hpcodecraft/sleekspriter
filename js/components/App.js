var App = React.createClass({
  render: function() {
    return (
      <div>
        <div className="area top"></div>
        <div className="area left">
          <ToolBox signal={this.props.signal} />
        </div>
        <div className="area center">
          <StageBox io={this.props.io}Â editor={this.props.editor} signal={this.props.signal} pixel={this.props.pixel}/>
        </div>
        <div className="area right">
          <LayerBox io={this.props.io} editor={this.props.editor} signal={this.props.signal} />
        </div>
        <div className="area bottom">
          <StatusBar editor={this.props.editor} signal={this.props.signal} />
        </div>
      </div>
    );
  },
  componentDidMount: function() {
    //console.log('mounted App');

    var self = this,
        subscriptions = [
          'gridToggled',
          'pixelSelected',
          'layerRemoved',
          'layerAdded',
          'layerSelected',
          'layerVisibilityChanged',
          'layerOpacityChanged',
          'layerNameChanged',
          'zoomChanged'
          //'frameContentChanged'
        ];

    subscriptions.forEach(function(item) {
      self.props.signal[item].add(self.updateProps);
    });


  },
  updateProps: function() {
    //console.log('updating App props');
    this.setProps({
      editor: editor,
      io: io
    });
  }

});