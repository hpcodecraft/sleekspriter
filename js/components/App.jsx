/** @jsx React.DOM */
var App = React.createClass({
  render: function() {

    var totalFrames = this.props.editor.frames.x * this.props.editor.frames.y,
        frames = [];

    for(var i=0; i < totalFrames; i++) frames[i] = i+1;

    return (
      <div id="App">
        <div className="area top">
          <ToolContainer editor={this.props.editor} />
        </div>
        <div className="area left">
          <ToolBox editor={this.props.editor} />
        </div>
        <div className="area center">
          <StageBox editor={this.props.editor} />
        </div>
        <div className="area right">
          <div id="layerboxhelper">
            <PreviewBox editor={this.props.editor} workspace={this.props.workspace} fold="preview" />
            <FrameBox editor={this.props.editor} workspace={this.props.workspace} fold="frames" />
          </div>
          <LayerBox editor={this.props.editor} workspace={this.props.workspace} fold="layers" />
        </div>
        <div className="area bottom">
          <StatusBar editor={this.props.editor} />
        </div>
        <div className="area offscreen">
          {frames.map(function(frame) {
            var id = 'OffscreenFrameCanvas-'+frame;
            return (
              <OffscreenFrameCanvas key={id} frame={frame} editor={this.props.editor} />
            );
          }, this)}

          <SelectionPattern editor={this.props.editor} />
        </div>
      </div>
    );
  },
  componentDidMount: function() {

    channel.subscribe('app.frame.select', this.updateProps);
    channel.subscribe('app.layer.select', this.updateProps);
    channel.subscribe('app.tool.select', this.updateProps);
    channel.subscribe('app.color.select', this.updateProps);
    channel.subscribe('app.pixel.select', this.updateProps);
    channel.subscribe('stage.grid.toggle', this.updateProps);
    channel.subscribe('stage.zoom.select', this.updateProps);

    channel.subscribe('file.layer.opacity.select', this.updateProps);
    channel.subscribe('file.layer.visibility.toggle', this.updateProps);
    return;

    //channel.subscribe('stage.grid.toggle', this.updateProps);
    //channel.subscribe('stage.zoom.select', this.updateProps);





    channel.subscribe('app.palette.select', this.updateProps);
    channel.subscribe('app.box.toggle', this.updateProps);


    channel.subscribe('app.brightnesstool.mode.select', this.updateProps);
    channel.subscribe('app.brightnesstool.intensity.select', this.updateProps);

    //channel.subscribe('app.layer.add', this.updateProps);


    channel.subscribe('file.layer.name.select', this.updateProps);



    channel.subscribe('stage.tool.move', this.updateProps);

  },
  updateProps: function() {
    //console.log('updating App props');
    this.setProps({editor: editor});
  },
  componentWillReceiveProps: function(props) {
    //console.log(props);
  }
});