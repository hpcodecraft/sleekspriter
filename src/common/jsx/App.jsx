var App = React.createClass({
  mixins: [PostalSubscriptionMixin],
  getInitialState: function() {
    return {
      tab: 'start',
      subscriptions: {
        'frame.select': this.updateProps,
        'layer.select': this.updateProps,
        'tool.select': this.updateProps,
        'color.select': this.updateProps,
        'palette.select': this.updateProps,
        'brightnesstool.mode.select': this.updateProps,
        'brightnesstool.intensity.select': this.updateProps,
        'background.select': this.updateProps,

        'settings.toggle': this.updateProps,
        'grid.toggle': this.updateProps,
        'zoom.select': this.updateProps,

        'layer.opacity.select': this.updateProps,
        'layer.visibility.toggle': this.updateProps,
        'layer.name.select': this.updateProps,
        'layer.add': this.updateProps,
        'layer.delete': this.updateProps,

        'animation.add': this.updateProps,
        'animation.delete': this.updateProps,
        'animation.select': this.updateProps,
        'animation.rename': this.updateProps,
        'animation.fps': this.updateProps,
        'animation.frame.add': this.updateProps,
        'animation.frame.delete': this.updateProps,

        'window.resize': this.updateProps,

        'screen.select': this.changeScreen,
      }
    }
  },
  render: function() {
    var tabs = [],
        windowClasses = {};

        if(this.state.tab !== 'start') {
          tabs.push('paint');
          if(this.props.editor.frames.total > 1) tabs.push('animate');
          tabs.push('export');
          tabs.push('debug');
        }

        windowClasses['window'] = true;
        windowClasses[this.state.tab] = true;
        windowClasses = React.addons.classSet(windowClasses);

    return (
      <div className="app">
        <nav className="menu" ref="menu">
          {this.state.tab === 'start' ? <label className="version">version @@version</label> :
            tabs.map(function(tab) {
              var label = tab[0].toUpperCase() + tab.substr(1, tab.length),
                  classes = React.addons.classSet({
                    tab: true,
                    active: this.state.tab === tab ? true : false,
                  });
              return (
                <div className={classes} key={tab} data-target={tab} onClick={this.selectTab}>{label}</div>
              )
            }, this)}
        </nav>
        <div className={windowClasses}>
          <ScreenStart />
          <ScreenPaint editor={this.props.editor} workspace={this.props.workspace} />
          <ScreenAnimate editor={this.props.editor} />
          <ScreenExport editor={this.props.editor} />
          <ScreenDebug editor={this.props.editor} />
        </div>
        <Modal editor={this.props.editor}Â />
      </div>
    );
  },
  updateProps: function() {
    this.setProps({editor: editor, workspace: workspace});
  },
  selectTab: function(event) {
    var target = event.target.getAttribute('data-target');
    channel.publish('screen.select', {target: target});
    this.setState({tab: target});
  },
  changeScreen: function(data) {
    this.setState({tab: data.target});
  },
});