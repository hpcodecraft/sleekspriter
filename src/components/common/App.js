import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import { throttle } from "lodash";

import MenuContainer from "../containers/MenuContainer";
import ModalContainer from "../containers/ModalContainer";
import ScreenPaintContainer from "../containers/ScreenPaintContainer";
import ScreenStartContainer from "../containers/ScreenStartContainer";
import ScreenExportContainer from "../containers/ScreenExportContainer";

import { ScreenBlocker } from "../screens";
import { Hotkeys } from "../../classes";

class App extends React.Component {
  componentWillReceiveProps(nextProps) {
    if (this.props.screen !== nextProps.screen)
      Hotkeys.unbind(this.props.screen);
  }

  componentDidMount() {
    window.addEventListener("resize", this.resize, false);
  }

  componentWillUnmount() {
    window.removeEventListener("resize", this.resize, false);
  }

  render() {
    Hotkeys.bind(this.props.screen); // TODO move this to componentDidUpdate

    const windowClasses = classnames({
      window: true,
      [this.props.screen]: true,
    });

    let activeScreen;
    switch (this.props.screen) {
      case "paint":
        activeScreen = <ScreenPaintContainer />;
        break;

      case "export":
        activeScreen = <ScreenExportContainer />;
        break;

      case "animate":
      case "start":
      default:
        activeScreen = <ScreenStartContainer />;
        break;
    }

    return (
      <div className="app">
        <MenuContainer />
        <div className={windowClasses}>{activeScreen}</div>
        <ModalContainer />
        <ScreenBlocker />
      </div>
    );
  }

  resize = throttle(() => {
    this.props.windowResize();
  }, 200);
}

App.propTypes = {
  screen: PropTypes.string.isRequired,
  windowResize: PropTypes.func.isRequired,
};

export default App;
