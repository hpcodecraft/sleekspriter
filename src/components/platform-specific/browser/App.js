import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import { throttle } from "lodash";

import MenuContainer from "../../containers/MenuContainer";
import ModalContainer from "../../containers/ModalContainer";
import ScreenPaintContainer from "../../containers/ScreenPaintContainer";
import ScreenExportContainer from "../../containers/ScreenExportContainer";

import { ScreenBlocker } from "../../screens";
import { Hotkeys } from "../../../classes";

import { setWindowTitle } from "../../../utils";

console.log(`Greetings! Running ${APPNAME} ${VERSION} in ${PLATFORM}`);

class App extends React.Component {
  componentDidUpdate(prevProps) {
    if (prevProps.screen !== this.props.screen) {
      Hotkeys.unbind(prevProps.screen);
      Hotkeys.bind(this.props.screen);
    }
  }

  componentDidMount() {
    window.addEventListener("resize", this.resize, false);
    Hotkeys.bind(this.props.screen);
    if (PLATFORM === "electron") {
      setWindowTitle(false, false);
    }
  }

  componentWillUnmount() {
    window.removeEventListener("resize", this.resize, false);
  }

  render() {
    const windowClasses = classnames({
      window: true,
      [PLATFORM]: true,
      [this.props.screen]: true,
    });

    let activeScreen;
    switch (this.props.screen) {
      case "paint":
        activeScreen = <ScreenPaintContainer />;
        break;

      case "export":
        activeScreen = <ScreenExportContainer />;
        break;
    }

    return (
      <div className="app">
        <MenuContainer />
        <div className={windowClasses}>{activeScreen}</div>
        <ModalContainer />
        <ScreenBlocker />
      </div>
    );
  }

  resize = throttle(() => {
    this.props.windowResize();
  }, 200);
}

App.propTypes = {
  screen: PropTypes.string.isRequired,
  windowResize: PropTypes.func.isRequired,
};

export default App;
