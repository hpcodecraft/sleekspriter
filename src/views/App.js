import React from 'react';
import ScreenPaint from './ScreenPaint';

const App = () => {
  return (
    <div className="app">
      <ScreenPaint />
    </div>
  );
};

export default App;
