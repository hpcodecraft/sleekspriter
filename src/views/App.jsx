import React from 'react';

const App = () => {
  return (
    <div>App, yep!</div>
  );
};

export default App;
