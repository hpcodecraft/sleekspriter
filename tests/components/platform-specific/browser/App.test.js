import App from "platform-specific/App";

// jest.mock("platform-specific/Menu", () => "Menu");

// jest.mock("../containers/MenuContainer", () => <div>Menu</div>);

describe("App", () => {
  const props = {
    screen: "paint",
    windowResize: jest.fn(),
  };

  it("should render correctly", () => {
    const wrapper = shallow(<App {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
