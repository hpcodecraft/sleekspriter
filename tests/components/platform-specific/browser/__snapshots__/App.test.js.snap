// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  className="app"
>
  <Connect(Menu) />
  <div
    className="window browser paint"
  >
    <Connect(ScreenPaint) />
  </div>
  <Connect(ModalContainer) />
  <ScreenBlocker />
</div>
`;
